code/network/lib/Loot/Network/ZMQ/Client.hs

```diff
@@ -433,75 +381,37 @@ reconnectPeers ZTNetCliEnv{..} nIds = liftIO $ do
 
         modifyTVar ztHeartbeatInfo $ \hbInfo ->
             foldl' (\m nId -> m & at nId %~ upd) hbInfo toReconnect
-        pure toReconnect
+        pure resources
 
     unless (null toReconnect) $
         ztLog ztCliLogging Warning $ "Reconnecting peers: " <> show nIds
-    forM_ toReconnect $ \nId -> do
-        Z.disconnect ztCliBack (ztNodeIdRouter nId)
-        Z.connect ztCliBack (ztNodeIdRouter nId)
-        Z.disconnect ztCliSub (ztNodeIdPub nId)
-        Z.connect ztCliSub (ztNodeIdPub nId)
-
--- This worker cleans up connection attempts that were stuck.
-connectionsWorker :: ZTNetCliEnv -> IO ()
-connectionsWorker ZTNetCliEnv{..} =
-    forever $ action `catchAny` handler
-  where
-    action = forever $ do
-        threadDelay 500000 -- 0.5 sec
-        curTime <- getCurrentTimeMs
-        let timeout = zsConnectingTimeout ztCliSettings
-        let leaveNode (_,(created,_,_))
-                | timeout == -1 = True
-                | otherwise = created + timeout > curTime
-        (disconnected,notConnSockets) <- atomically $ do
-            prev <- readTVar (ztPeers ^. pivConnecting)
-            let (stay,leave) = L.partition leaveNode (HMap.toList prev)
-            writeTVar (ztPeers ^. pivConnecting) (HMap.fromList stay)
-            pure (leave,map (\(_, (_,sock,_)) -> sock) stay)
-        -- Sending messages until connected/disconnected. This
-        -- prevents the situation when the server starts later then
-        -- the client, in which server doesn't receive our first
-        -- handshake request.
-        forM_ notConnSockets $ \dSock ->
-            Z.sendMulti dSock $ NE.fromList ["",tag_getId]
-        unless (null disconnected) $ do
-            ztLog ztCliLogging Warning $
-                "Connecting attempt timeouted for peers: " <>
-                show (map fst disconnected)
-            forM_ disconnected $ \(_,(_,sock,adapter)) ->
-                Z.close sock >> adapterRelease adapter
-    handler e = do
-        ztLog ztCliLogging Error $
-            "Connections timeout worker failed: " <> show e <>
-            ", restarting in 2s"
-        threadDelay 2000000
+    forM_ toReconnect $ \(nId, PeerRes{..}) -> do
+        Z.disconnect prBack (ztNodeIdRouter nId)
```

Comment:
honestly I would prefer to disconnect both and then connect both and do it atomically to avoid inconsistency between two sockets. thought, it's not so important on current stage of development, IMHO

Reply:
I still think that reconnecting peers is something we should get rid of, implicitly offloading it to the ZMQ backend.